test/hotspot/jtreg/serviceability/jvmti/GetOwnedMonitorInfo/GetOwnedMonitorInfoTest.java

```diff
@@ -46,19 +46,40 @@ public class GetOwnedMonitorInfoTest {
         }
     }
 
+    private static native void jniMonitorEnter(Object obj);
     private static native int check();
     private static native boolean hasEventPosted();
 
+    private static void jniMonitorEnterAndLetObjectDie() {
+        // The monitor iterator used by GetOwnedMonitorInfo used to
+        // assert when an owned monitor with a dead object was found.
+        // Inject this situation into this test that performs other
+        // GetOwnedMonitorInfo testing.
+        Object obj = new Object() { public String toString() {return "";} };
+        jniMonitorEnter(obj);
+        if (!Thread.holdsLock(obj)) {
+            throw new RuntimeException("The object is not locked");
+        }
+        obj = null;
+        System.gc();
+    }
+
     public static void main(String[] args) throws Exception {
-        runTest(true);
-        runTest(false);
+        runTest(true, true);
+        runTest(true, false);
+        runTest(false, true);
+        runTest(false, false);
     }
 
-    public static void runTest(boolean isVirtual) throws Exception {
+    public static void runTest(boolean isVirtual, boolean jni) throws Exception {
         var threadFactory = isVirtual ? Thread.ofVirtual().factory() : Thread.ofPlatform().factory();
         final GetOwnedMonitorInfoTest lock = new GetOwnedMonitorInfoTest();
 
         Thread t1 = threadFactory.newThread(() -> {
```

Comment:
Pre-existing nit: by default virtual threads have no name, so the output in the virtual thread case looks a little odd. Can you add:
```
Thread.currentThread().setName("Worker-Thread");
```
please.

Reply:
Sure.